src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -243,27 +237,30 @@ protected function generateAPIKey(Document $project): Document
      * @throws Structure
      * @throws \Utopia\Database\Exception
      */
-    protected function processMigration(Document $project, Document $migration, Log $log): void
+    protected function processMigration(Document $project, Document $group, Log $log): void
     {
         /**
-         * @var Document $migrationDocument
+         * @var Document $migration
          * @var Transfer $transfer
          */
-        $migrationDocument = null;
+        $groupDocument = null;
         $transfer = null;
         $projectDocument = $this->dbForConsole->getDocument('projects', $project->getId());
         $tempAPIKey = $this->generateAPIKey($projectDocument);
 
         try {
-            $migrationDocument = $this->dbForProject->getDocument('migrations', $migration->getId());
-            $migrationDocument->setAttribute('stage', 'processing');
-            $migrationDocument->setAttribute('status', 'processing');
+            $group = $this->dbForProject->getDocument('groupMigrations', $group->getId());
+
+            $migration = $this->dbForProject->getDocument('migrations', $group->getAttribute('migrationId', ''));
+            $migration->setAttribute('status', 'processing');
+
             $log->addBreadcrumb(new Breadcrumb("debug", "migration", "Migration hit stage 'processing'", \microtime(true)));
-            $this->updateMigrationDocument($migrationDocument, $projectDocument);
 
-            $log->addTag('type', $migrationDocument->getAttribute('source'));
+            $this->updateMigrationDocument($migration, $projectDocument);
+
+            $log->addTag('type', $migration->getAttribute('source'));
```

Comment:
Lets group the logger related statements together and the migrations related statements together